client-v2/src/main/java/com/clickhouse/client/api/internal/HttpAPIClientHelper.java

```diff
@@ -583,7 +583,7 @@ public boolean shouldRetry(Exception ex, Map<String, Object> requestSettings) {
             return retryCauses.contains(ClientFaultCause.NoHttpResponse);
         }
 
-        if (ex instanceof ConnectException) {
+        if (ex instanceof ConnectTimeoutException) {

```

Comment:
I think they should be both: 
- ConnectException - means that socket failed to connect. When we implement TCP protocol support we might use this exception 
- org.apache.hc.client5.http.ConnectTimeoutException is child of java.net.SocketTimeoutException - means that throws when read or accept (according to the JavaDoc of the last one). And ConnectTimeoutException is client specific exception. It will not be thrown by other client.

Reply:
> ConnectException - means that socket failed to connect. When we implement TCP protocol support we might use this exception

I agree that this is a useful exception to check for, but this particular if block then proceeds to check `retryCauses.contains(ClientFaultCause.ConnectTimeout)`, and so `ConnectException` doesn't seem applicable (to `ClientFaultCause.ConnectTimeout`)

Reply:
It is applicable - because ClientFaultCause.ConnectTimeout is about any error while connection initiation. 
ConnectException is all about it:
```
/**
 * Signals that an error occurred while attempting to connect a
 * socket to a remote address and port.  Typically, the connection
 * was refused remotely (e.g., no process is listening on the
 * remote address/port).
 *
 * @since   1.1
 */
public class ConnectException extends SocketException 
```